controllers/storagecluster/external_resources.go

```diff
@@ -552,7 +523,7 @@ func (r *StorageClusterReconciler) createExternalModeStorageClasses(sccs []Stora
 			if existing.DeletionTimestamp != nil {
 				return fmt.Errorf("failed to restore StorageClass  %s because it is marked for deletion", existing.Name)
 			}
-			if !reflect.DeepEqual(sc.Parameters, existing.Parameters) {
+			if !reflect.DeepEqual(sc.Parameters, existing.Parameters) || existing.Labels[util.ExternalClassLabelKey] == "" {
```

Comment:
shouldn't we check if the existing label value is != what we desire?

Reply:
We desire to update the storageclass when the existing CR doesn't have the desired external labels,
So if `existing.Labels[util.ExternalClassLabelKey] == ""` means we need to update it

Reply:
What if the customer updates the label in external storageClass to existing.Labels[util.ExternalClassLabelKey] = "false"? This code will see that we have a label and it won't update the StorageClass and consumer controller will pick this up for distribution as it won't be filtered out

Reply:
If the customer update it manually, code will again set it to `true`.